ably-ios/ARTAuth.m

```diff
@@ -38,12 +40,17 @@ - (instancetype)init:(ARTRest *)rest withOptions:(ARTClientOptions *)options {
         } else if (options.tokenDetails) {
             [self.logger debug:@"ARTAuth: setting up auth method Token with supplied token only"];
             _method = ARTAuthMethodToken;
+        } else if (options.authUrl && options.authCallback) {
+            [NSException raise:@"ARTAuthException" format:@"Could not setup authentication method with given options."];
```

Comment:
Sure, I will change it. The first PR to merge is the conversion into Swift 2.

About the `clientId`... the attribute is missing on Auth class. I don't know why and I'm trying to figure it out.

On 12 October 2015 at 22:04:20, mattheworiordan (notifications@github.com) wrote:
In ably-ios/ARTAuth.m:

> @@ -38,12 +40,17 @@ - (instancetype)init:(ARTRest *)rest withOptions:(ARTClientOptions *)options {
>          } else if (options.tokenDetails) {
>              [self.logger debug:@"ARTAuth: setting up auth method Token with supplied token only"];
>              _method = ARTAuthMethodToken;
> +        } else if (options.authUrl && options.authCallback) {
> +            [NSException raise:@"ARTAuthException" format:@"Could not setup authentication method with given options."];
Can this rather say "Incompatible authentication configuration: please specify either authCallback and authUrl"

—
Reply to this email directly or view it on GitHub.